platform/linux-generic/odp_crypto_ipsecmb.c

```diff
@@ -0,0 +1,999 @@
+
+	op_result = get_op_result_from_packet(pkt);
+	op_result->cipher_status.alg_err = cipher_err;
+	op_result->cipher_status.hw_err = ODP_CRYPTO_HW_ERR_NONE;
+	op_result->auth_status.alg_err = auth_err;
+	op_result->auth_status.hw_err = ODP_CRYPTO_HW_ERR_NONE;
+	op_result->ok = (cipher_err == ODP_CRYPTO_ALG_ERR_NONE &&
+			 auth_err   == ODP_CRYPTO_ALG_ERR_NONE);
+}
+
+static inline void set_crypto_op_result_ok(odp_packet_t pkt)
+{
+	set_crypto_op_result(pkt,
+			     ODP_CRYPTO_ALG_ERR_NONE,
+			     ODP_CRYPTO_ALG_ERR_NONE);
+}
+
+static void
+null_crypto_routine(odp_packet_t pkt ODP_UNUSED,
+		    const odp_crypto_packet_op_param_t *param ODP_UNUSED,
+		    odp_crypto_generic_session_t *session ODP_UNUSED)
+{
+	set_crypto_op_result_ok(pkt);
+}
+
+
+static
+void zuc_eea3_cipher_op(odp_packet_t pkt,
+		      const odp_crypto_packet_op_param_t *param,
+		      odp_crypto_generic_session_t *session)
+{
+	IMB_MGR *mb_mgr = local.mb_mgr;
+	uint8_t *iv_ptr;
+	uint32_t in_pos = param->cipher_range.offset;
+	uint32_t in_len = param->cipher_range.length;
+        uint8_t dst_data[MAX_BUFFER_LENGTH_IN_BYTES];
+
+#if ODP_DEPRECATED_API
+	if (param->cipher_iv_ptr)
+		iv_ptr = param->cipher_iv_ptr;
+	else if (session->p.cipher_iv.data)
+		iv_ptr = session->cipher.iv_data;
+	else {
+		set_crypto_op_result(pkt,
+				     ODP_CRYPTO_ALG_ERR_DATA_SIZE,
+				     ODP_CRYPTO_ALG_ERR_NONE);
+		return;
+	}
+#else
+	iv_ptr = param->cipher_iv_ptr;
+	_ODP_ASSERT(session->p.cipher_iv_len == 0 || iv_ptr != NULL);
+#endif
+
+	uint32_t seg_len = 0;
+	uint8_t *data = odp_packet_offset(pkt, in_pos, &seg_len, NULL);
+
+	if (odp_unlikely(odp_packet_is_segmented(pkt))) {
+		/* Packet is segmented within the cipher range. Copy the cipher
+		 * range to a contiguous buffer. */
+		odp_packet_copy_to_mem(pkt, in_pos, in_len, local.buffer);
+
+		data = local.buffer;
+	}
+
+	if (session->p.cipher_key.length == 16) {
+		/* ZUC128 EEA3 */
+		IMB_ZUC_EEA3_1_BUFFER(mb_mgr, (const void  **)session->cipher.key_data,
+				      (const void *)iv_ptr,
+				      (const void *)data,
+				      (void *)dst_data,
+                                      in_len);
+	} else if (session->p.cipher_key.length == 32) {
+		/* ZUC256 EEA3 */
+		IMB_ZUC256_EEA3_1_BUFFER(mb_mgr, (const void **)session->cipher.key_data,
+					 (const void *)iv_ptr,
+					 (const void *)data,
+					 (void *)dst_data,
+					 in_len);
+	}
+
+	odp_packet_copy_from_mem(pkt, in_pos, in_len, dst_data);
+
+	set_crypto_op_result_ok(pkt);
+}
+
+static int process_zuc_eea3_param(odp_crypto_generic_session_t *session)
+{
+	/* Verify key length is valid */
+	if (16 != session->p.cipher_key.length &&
+	    32 != session->p.cipher_key.length)
+		return -1;
+
+	/* Verify the IV length is correct */
+	if (16 != session->p.cipher_iv_len &&
+	    25 != session->p.cipher_iv_len)
+		return -1;
+
+	memcpy(session->cipher.key_data, session->p.cipher_key.data,
+		session->p.cipher_key.length);
+
+	/* Set function */
```

Comment:
> /* Verify key length is valid */
> /* Verify the IV length is correct */
> /* Set function */

These kind of comments are not useful since the code is obvious anyway. The same goes for many other comments in the code.

Reply:
Sound good, removed all the unnecessary comments.